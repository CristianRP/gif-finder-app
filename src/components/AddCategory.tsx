import { ChangeEvent, Dispatch, FormEvent, SetStateAction, useState } from 'react'

interface AddCategoryProps {
  setCategories: Dispatch<SetStateAction<string[]>>;
}

export const AddCategory = ( { setCategories }: AddCategoryProps) => {

  const [ inputValue, setInputValue ] = useState('');

  const onInputChange = ( { target }: ChangeEvent<HTMLInputElement>) => {
    setInputValue( target.value );
  }

  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    if (inputValue.trim().length <= 1) return;

    setCategories( (categories: string[]) => [ inputValue, ...categories ] );
    setInputValue('');
  }

  return (
    <form onSubmit={ onSubmit }>
      <input
        type="text"
        placeholder='Search Gif'
        value={inputValue}
        onChange={ onInputChange }
      />
    </form>
  )
}

