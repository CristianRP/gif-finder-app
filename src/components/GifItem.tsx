import PropTypes from 'prop-types';

import { GifResponse } from '../helpers/getGifs'

export const GifItem = ({ title, url }: GifResponse) => {
  return (
    <div className="card--item">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
