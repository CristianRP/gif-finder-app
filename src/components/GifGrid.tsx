import { FC } from 'react';
import PropTypes from 'prop-types';

import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

type GifGridProps = {
  category: string;
}

export const GifGrid: FC<GifGridProps> = ({ category }) => {

  const { images } = useFetchGifs( category );
  
  return (
    <>
      <h3>{ category }</h3>
      <div className='card-grid'>
        {
          images.map( image => (
            <GifItem key={ image.id } { ...image } />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
