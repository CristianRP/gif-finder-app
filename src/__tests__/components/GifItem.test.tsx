import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components';

describe('Tests on <GifItem /> component', () => {

  const img = {
    id: new Date().toISOString(),
    title: 'Test',
    url: 'http://images.com/image.png'
  }

  test('should match snapshot', () => {
    const { container } = render( <GifItem { ...img } /> );

    expect( container ).toMatchSnapshot();
  });

  test('should show the image with the URL and ALT', () => { 
    render( <GifItem {...img} />)
    // screen.debug()
    // expect(screen.getByRole<HTMLImageElement>('img').src).toBe( img.url );
    // expect(screen.getByRole<HTMLImageElement>('img').alt).toBe( img.url );
    const { src, alt } = screen.getByRole<HTMLImageElement>('img');
    expect( src ).toBe( img.url );
    expect( alt ).toBe( img.title );
  });

  test('should show the title in the component', () => {
    render( <GifItem {...img} /> );
    expect( screen.getByText( img.title ) ).toBeTruthy();
  });

});
