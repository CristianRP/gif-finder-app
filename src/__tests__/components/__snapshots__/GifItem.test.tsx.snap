// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> component should match snapshot 1`] = `
<div>
  <div
    class="card--item"
  >
    <img
      alt="Test"
      src="http://images.com/image.png"
    />
    <p>
      Test
    </p>
  </div>
</div>
`;
